File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/plugin/req/Req.kt
package com.mytiki.sdk.capture.receipt.capacitor.plugin.req

import com.getcapacitor.PluginCall
import java.lang.Exception

open class Req(call: PluginCall){
    val requestId: String
    init{
        val reqId = call.getString("requestId")
        if(reqId == null){
            call.reject("Add a requestId in the call.")
            throw(Exception("Add a requestId in the call."))
        }
        requestId = reqId
    }
}

File: ios/Plugin/Plugin/Req/Req.swift
/*
 * Req Class
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in the root directory.
 */

import Foundation
import Capacitor

/// A Swift class for handling request data in a Capacitor plugin.
public class Req {
    let requestId: String

    
    /// Initializes the Req class with the request identifier from a CAPPluginCall.
    ///
    /// - Parameter call: The CAPPluginCall object representing the request.
    /// - Throws: An error if the request identifier is missing.
    init(_ call: CAPPluginCall) throws {
        guard let reqId = call.getString("requestId") else {
            call.reject("Add a requestId in the call.")
            throw NSError(domain: "", code: 0, userInfo: [NSLocalizedDescriptionKey: "Add a requestId in the call."])
        }
        requestId = reqId
    }
}
